### DISCUSSION TITLE
Does lazer replace already imported stable maps (from previous lazer version) with hard links?

### DISCUSSION THREAD

**Discussion by LittleEndu** [ppy/osu#21827] [Q&A] [answered]
I started importing beatmaps from stable, and my hard disk usage has already fallen by 2 GB. I assume this is because old files are getting replaced with hard links, but that part specifically isn't mentioned in the wiki.

**Comment by bdach** [comment, marked as answer]
That should not be happening.

https://github.com/ppy/osu/blob/b4a3f872ee73c40a8f25356efca2eae2ba382e3b/osu.Game/Database/RealmFileStore.cs#L47-L56

Note that `copyToStore()` is invoked only if the target file doesn't exist or doesn't match hash.

**Reply by LittleEndu** [reply]
I guess the 2 GB must have been some temporary files from other programs clearing up then. Thanks for the answer

**Reply by peppy** [reply]
That said, we *could* potentially make this work. It might be more user-friendly if it did as proposed.
